### PR TITLE
[Codecov] Install aarch64 codecov on arm runners

### PR DESCRIPTION
**Pull request by amenasria**

### CODE REVIEW

File: setup_codecov.sh [hunk 1]
```diff
@@ -4,11 +4,34 @@ set -euo pipefail
...
+# Check if the architecture was provided
+if [ -z "$1" ]; then
+  echo "Usage: $0 <architecture>"
+  echo "Example: $0 arm64"
+  exit 1
+fi
+
+ARCH=$1
+
+case $ARCH in
```
Comment (FlorentClarret) [inline setup_codecov.sh:16]: Just an idea but we already have `DD_TARGET_ARCH` in every docker file (e.g. [here](https://github.com/DataDog/datadog-agent-buildimages/blob/f6a5fc0c98e06790844e0d2e0c9c64caceefa7bb/deb-arm/Dockerfile#L26))

Would it make sense to use it as a parameter to have the exact same thing in all the dockerfiles and handle the possible values in the script?

**Review by FlorentClarret — COMMENTED** [review]
Left one question, but otherwise LGTM

File: setup_codecov.sh (hunk 1 above)
Comment (FlorentClarret) [reply setup_codecov.sh:16]: (That's also an env variable)

**Review by FlorentClarret — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

File: setup_codecov.sh [hunk 2]
```diff
@@ -1,14 +1,37 @@
...
+  echo "DD_TARGET_ARCH environment variable is not set. The Codecov x64 uploader will be installed by default."
+  CODECOV_ARCH="linux"
+else
+  case $DD_TARGET_ARCH in
+    aarch64)
+      CODECOV_ARCH="aarch64"
+      ;;
+    armhf)
+      CODECOV_ARCH="aarch64"
+      ;;
```
Comment (FlorentClarret) [inline setup_codecov.sh:18]:
```suggestion
    aarch64|armhf)
      CODECOV_ARCH="aarch64"
      ;;
```

IIRC, but I'm a bit rusted so try it before :)

File: setup_codecov.sh (hunk 1 above)
Comment (amenasria) [reply setup_codecov.sh]: Good point thanks ! Fixed in [7823589](https://github.com/DataDog/datadog-agent-buildimages/pull/589/commits/7823589373f3deba6678c3d7c74168a3d1fd1220).

File: setup_codecov.sh (hunk 2 above)
Comment (amenasria) [reply setup_codecov.sh:18]: Nice ! Didn't know this syntax ! Thank you

File: setup_codecov.sh [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
+  case $DD_TARGET_ARCH in
+    aarch64|armhf)
+      CODECOV_ARCH="aarch64"
+      ;;
+    x64)
+      CODECOV_ARCH="linux"
+      ;;
+    *)
+      echo "Invalid DD_TARGET_ARCH value: ${DD_TARGET_ARCH}"
+      echo "The DD_TARGET_ARCH values supported by the Codecov setup are: aarch64, x64"
```
Comment (FlorentClarret) [inline setup_codecov.sh:21]:
```suggestion
      echo "The DD_TARGET_ARCH values supported by the Codecov setup are: aarch64, x64 and armhf"
```

🙈

File: setup_codecov.sh [hunk 4]
```diff
@@ -1,14 +1,34 @@
 #!/bin/bash
 
-set -euo pipefail
+set -eo pipefail
 
 CODECOV_VERSION=0.6.1
 
+# Determine the Architecture to install the Codecov uploader.
```
Comment (CelianR) [inline setup_codecov.sh:7]:
```suggestion
# Determine the Architecture to install the Codecov uploader
```

File: setup_codecov.sh [hunk 5]
```diff
@@ -1,14 +1,34 @@
...
+  case $DD_TARGET_ARCH in
+    aarch64|armhf)
+      CODECOV_ARCH="aarch64"
+      ;;
+    x64)
+      CODECOV_ARCH="linux"
+      ;;
+    *)
+      echo "Invalid DD_TARGET_ARCH value: ${DD_TARGET_ARCH}"
+      echo "The DD_TARGET_ARCH values supported by the Codecov setup are: aarch64, x64 and armhf"
```
Comment (CelianR) [inline setup_codecov.sh:21]:
```suggestion
      echo "Invalid DD_TARGET_ARCH value: ${DD_TARGET_ARCH}" >& 2
      echo "The DD_TARGET_ARCH values supported by the Codecov setup are: aarch64, x64 and armhf" >& 2
```

File: setup_codecov.sh [hunk 6]
```diff
@@ -1,14 +1,34 @@
...
+fi
+
 # Integrity checking the uploader
 curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import
-curl -Os https://uploader.codecov.io/v${CODECOV_VERSION}/linux/codecov
-curl -Os https://uploader.codecov.io/v${CODECOV_VERSION}/linux/codecov.SHA256SUM
-curl -Os https://uploader.codecov.io/v${CODECOV_VERSION}/linux/codecov.SHA256SUM.sig
+curl -Os https://uploader.codecov.io/v${CODECOV_VERSION}/${CODECOV_ARCH}/codecov
+curl -Os https://uploader.codecov.io/v${CODECOV_VERSION}/${CODECOV_ARCH}/codecov.SHA256SUM
+curl -Os https://uploader.codecov.io/v${CODECOV_VERSION}/${CODECOV_ARCH}/codecov.SHA256SUM.sig
```
Comment (CelianR) [inline setup_codecov.sh:31]:
```suggestion
curl -Os https://uploader.codecov.io/v${CODECOV_VERSION}/$CODECOV_ARCH/codecov
curl -Os https://uploader.codecov.io/v${CODECOV_VERSION}/$CODECOV_ARCH/codecov.SHA256SUM
curl -Os https://uploader.codecov.io/v${CODECOV_VERSION}/$CODECOV_ARCH/codecov.SHA256SUM.sig
```

File: setup_codecov.sh [hunk 7]
```diff
@@ -1,14 +1,34 @@
 #!/bin/bash
 
-set -euo pipefail
+set -eo pipefail
 
 CODECOV_VERSION=0.6.1
 
+# Determine the Architecture to install the Codecov uploader.
+if [[ -z "${DD_TARGET_ARCH}" ]]; then
```
Comment (CelianR) [inline setup_codecov.sh:8]:
```suggestion
if [ -z "$DD_TARGET_ARCH" ]; then
```

**Review by CelianR — APPROVED** [review]

**Pull request merged**